src/vanilla.ts

```diff
@@ -10,10 +22,35 @@ export type PartialState<
 > =
   | (Pick<T, K1> | Pick<T, K2> | Pick<T, K3> | Pick<T, K4> | T)
   | ((state: T) => Pick<T, K1> | Pick<T, K2> | Pick<T, K3> | Pick<T, K4> | T)
+
+/**
+ * @deprecated Use `(t: T) => U` instead of `StateSelector<T, U>`.
+ * `StateSelector` will be removed in next major.
+ */
 export type StateSelector<T extends State, U> = (state: T) => U
+
+/**
+ * @deprecated Use `(a: T, b: T) => boolean` instead of `EqualityChecker<T>.
+ * `EqualityChecker` will be removed in next major.
+ */
 export type EqualityChecker<T> = (state: T, newState: T) => boolean
+
+/**
+ * @deprecated Use `(state: T, prevState: T) => void` instead of `StateListener<T>`.
+ * `StateListener` will be removed in next major.
+ */
 export type StateListener<T> = (state: T, previousState: T) => void
```

Comment:
For `StateSelector`, `EqualityChecker` and `StateListener`, I'm not sure what will happen. I guess many users use them. They might not be necessary nor best named, but not too confusing. So, for now I lean to keep them.

Reply:
I can try there three actually, and see if someone gets confused.

Reply:
> I can try there three actually, and see if someone gets confused.

Did you thought about this and decided that if you want to keep them or not?

Reply:
Let's keep them.
I saw some tweets and codesandboxes and know people use them.
If we were to remove those, it would be v5 (taking migration period longer like a few months to a year), which will not happen any time soon.